pkg/validation-webhook/certwatcher.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"crypto/tls"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"k8s.io/klog"
+)
+
```

Comment:
Is it because our webhook is not generated by controller-gen, so that we can't directly use "github.com/kubernetes-sigs/controller-runtime/pkg/webhook/internal/certwatcher"?

If so, please add comments in the beginning of this file.

Reply:
We cannot import github.com/kubernetes-sigs/controller-runtime/pkg/webhook/internal/certwatcher because it is an internal package:
https://golang.org/doc/go1.4#internalpackages

Added a comment about the origin of this file in certwatcher.go